deps/obs-scripting/obslua/CMakeLists.txt

```diff
@@ -25,26 +25,28 @@ swig_add_library(
 
 add_library(OBS::lua ALIAS obslua)
 
-target_compile_options(obslua PRIVATE "$<$<C_COMPILER_ID:AppleClang,Clang>:-Wno-error=shorten-64-to-32>")
+target_compile_options(
+  obslua
+  PRIVATE $<$<PLATFORM_ID:Windows>:/wd4054>
+          $<$<PLATFORM_ID:Windows>:/wd4197>
+          $<$<PLATFORM_ID:Windows>:/wd4244>
+          $<$<PLATFORM_ID:Windows>:/wd4267>
+          $<$<C_COMPILER_ID:AppleClang,Clang>:-Wno-shorten-64-to-32>
+          $<$<C_COMPILER_ID:AppleClang,Clang>:-Wno-unreachable-code>
+          $<$<C_COMPILER_ID:GNU>:-Wno-maybe-uninitialized>)
```

Comment:
At least in my checks GCC was the only one who emitted that warning.

Reply:
To clarify, I asked this because the section it replaced was:
```
elseif(OS_LINUX OR OS_FREEBSD)
  target_compile_options(obslua PRIVATE -Wno-maybe-uninitialized)
```

So I was wondering if it's safe to assume that only GCC is used on those systems (or do people sometimes use clang on those?) _and_ that it's the only compiler (that we cover) that emits this warning. If GCC is the only compiler that emits the warning, then this is fine.

Reply:
The `maybe-uninitialized` warning is not supported by `clang` (it's a GCC-only warning) and was present in the old code. It does not seem to be emitted by the code generated by `swig` anymore, but if we wish to retain it, it needs to be used for GCC only.

Reply:
Got it.